### DISCUSSION TITLE
Invalid undefined affter mutation for update cache

### DISCUSSION THREAD

**Discussion by edw19** [urql-graphql/urql#1902] [Q&A] [answered]
i have this code with urql for update cache 
![image](https://user-images.githubusercontent.com/47750859/131223484-4affc8b1-d0f8-4415-8aee-4cced0b8acdd.png)

but at the moment execute mutation  i have this warning and update cache dont working

![image](https://user-images.githubusercontent.com/47750859/131223513-b945792c-565c-4170-8dd3-e64eede16324.png)

what am i doing wrong?

**Comment by kitten** [comment, marked as answer]
This is easily missed, however `Array#push` has a return value, which is the length of the array after the push. Since you're returning that, Graphcache sees that as your data which isn't "allowed", so to speak. Instead, it expects you to return `data`, either after mutating it as you're doing, or after you've made a copy of it.
